### DISCUSSION TITLE
Allow users to add syntax highlightable languages

### DISCUSSION THREAD

**Discussion by ernix** [growilabs/growi#7748] [Ideas / Requests]
Environment
------------

### Host

| item     | version |
| ---      | --- |
|OS        ||
|GROWI     |6.0.3|
|node.js   |14.21.2|
|npm       |6.14.17|
|yarn      |1.22.19|
|Using Docker|yes/no|
|Using [growi-docker-compose][growi-docker-compose]|yes/no|

[growi-docker-compose]: https://github.com/weseek/growi-docker-compose

*(Accessing https://{GROWI_HOST}/admin helps you to fill in above versions)*

### Client

| item     | version |
| ---      | --- |
|OS        |M2 Mac Ventura|
|browser   |Chrome 109.0.5414.119|

How to reproduce? (再現手順)
---------------------------

1. Write VBScript code block (VBScriptのコードブロックを作成します)
```
~~~vb:test.vbs
Dim strFoo
Dim I

For I = 0 To 10
  If I > 5 Then
    Exit For
  End If
Next
~~~
```

What happens? (症状)
---------------------

- No syntax highlighting after edit pages (ページにシンタックスハイライトが反映されません)
- Editor pane somehow gets syntax highlighting (なぜかエディターの方はハイライトされているようです)

https://demo.growi.org/63da645b509f5e8391799b54

<img width="1029" alt="Screenshot 2023-02-01 at 21 58 21" src="https://user-images.githubusercontent.com/343556/216051779-69fc8947-3bd8-4205-bbf8-01424e181520.png">

What is the expected result? (期待される動作)
-------------------------------------------

- Code blocks get syntax highlighted (シンタックスハイライトが有効になることを期待します)

**Comment by yuki-takei** [comment]
@ernix 
こちらはバグではなくリクエストとして Discussions に移します。

## Summary
- エディタのシンタックスハイライトは CodeMirror、HTML の方は [react-syntax-highlighter](https://github.com/react-syntax-highlighter/react-syntax-highlighter) で別の実装
    - サポートしている言語は [AVAILABLE_LANGUAGES_PRISM.MD](https://github.com/react-syntax-highlighter/react-syntax-highlighter/blob/15.5.0/AVAILABLE_LANGUAGES_PRISM.MD)
    - VBScript に限れば `vbnet` で代用可能かもしれない(ただしその場合エディタ側はハイライトされなくなる)
- v6.x で Custom HTML Header をセットする機能が失われたため、現状シンタックスハイライトの言語やテーマの種類を増やす手段がない
